analyzer/main.zeek

```diff
@@ -289,7 +289,7 @@ function set_session(c: connection, message_id: int, opcode: LDAP::ProtocolOpcod
 @if (Version::at_least("4.2.0"))
 event analyzer_confirmation(c: connection, atype: Analyzer::Tag, aid: count) {
 @else
-event protocol_confirmation(c: connection, atype: AllAnalyzers::Tag, aid: count) {
+event protocol_confirmation(c: connection, atype: Analyzer::Tag, aid: count) {
 @endif
```

Comment:
I'm loosing track where we even started here, but it's still not quite right. I think this is what we want:

```
@if (Version::at_least("4.2.0"))
event analyzer_confirmation(c: connection, atype: AllAnalyzers::Tag, aid: count)
@else
event protocol_confirmation(c: connection, atype: Analyzer::Tag, aid: count) {
@endif
```

That said, I'm now wondering if we have a problem in Zeek (or whether I'm just confused at this point): Here are the prototypes in current `master`: 

```
# git grep '^event.*_confirmation' src/event.bif
src/event.bif:event analyzer_confirmation%(c: connection, atype: Analyzer::Tag, aid: count%);
src/event.bif:event protocol_confirmation%(c: connection, atype: AllAnalyzers::Tag, aid: count%) &deprecated="Remove in v5.1. Use analyzer_confirmation.";
```

At the same time, we're using the new event like this in `master`:

```
# git grep '^event.*_confirmation' scripts/
scripts/base/frameworks/dpd/main.zeek:event analyzer_confirmation(c: connection, atype: AllAnalyzers::Tag, aid: count) &priority=10
scripts/base/protocols/conn/inactivity.zeek:event analyzer_confirmation(c: connection, atype: AllAnalyzers::Tag, aid: count)
...
```

In 4.1 we have:

```
# git grep '^event.*_confirmation' scripts/
scripts/base/frameworks/dpd/main.zeek:event protocol_confirmation(c: connection, atype: Analyzer::Tag, aid: count) &priority=10
scripts/base/protocols/conn/inactivity.zeek:event protocol_confirmation(c: connection, atype: Analyzer::Tag, aid: count)
...
```

I think usage is right, but the prototypes look reversed. Paging @timwoj .

Reply:
`analyzer_*` should be using `AllAnalyzers` everywhere. If there are spots where it isn't, then I probably screwed something up in that migration 😅

Reply:
The way around this here seems to be to use `Analyzer::Tag` everywhere. I have filed zeek/zeek#2261 for any needed follow-up on the Zeek side.